### PR TITLE
 Restore order of operators before executing the git command only for < py3.6

### PR DESCRIPTION
**Pull request by bertwesarg**
Since Python 3.6 kwargs are stable ordered again and Python 3.5 reached
its end-of-life too, so we can revert 89ade7bfff534ae799d7dd693b206931d5ed3d4f
again. Thus make it able to pass ordered options to Git commands again.

### CODE REVIEW

File: .travis.yml
Comment (Byron) [inline .travis.yml:41]: I think this should check for `'3.6'` instead to not loose documentation validation.

**Review by Byron — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution, it's much appreciated!

Does that also mean that kwargs are now ordered by default? Or does it mean that the order won't change between calls even though the order is not as specified?
In the latter case, the result might still be considered unstable compared to ordering them.

What do you think?

**Comment by bertwesarg** [comment]
> What do you think?

The PEP speaks about _preserving order_, this should also mean, that the order wont change _between calls_. I cannot reproduce the example `git.remote.pull()`, given in the original commit. But I can reproduce the _random_ order of kwargs between two executions of Python 3.5 with this code:

```python
def pkwargs(**kwargs):
    print(kwargs)
pkwargs(**{'one': 1, 'two': 2, 'three': 3, 'four': 4})
```

This gives me one of these possible outputs:

```
{'three': 3, 'four': 4, 'one': 1, 'two': 2}
{'two': 2, 'one': 1, 'three': 3, 'four': 4}
```

For Python 3.6 it is stable and preserves the order:

```
{'one': 1, 'two': 2, 'three': 3, 'four': 4}
```

**Comment by bertwesarg** [comment]
The PEP link is also in the commit message: https://www.python.org/dev/peps/pep-0468/

**Comment by Byron** [comment]
Thanks a lot! Now this PR would be ready for merge even though I feel a little uncomfortable to remove support half a year after py3.5 reached its end of life given that it took 2 years to remove 3.4 support to be able to support inline types which has incredibly high value.

My suggestion is to wait with the merge till Sept. 5, 2020 giving folks another half a year to stop using python 3.5 or face the potential for breakage.

If you would like a merge right now one could add a python version check to keep the sort code for py3.5 maybe along a depreciation warning to let people know the date of removal.

Thanks for sharing your thoughts.

**Comment by bertwesarg** [comment]
I don't think, shifting the date further back will have any effect. Most of us (the users) will drop py3.5, ony if we notice, that its not supported by the package we are using. But in the end, this is your call.

I also do not have a problem, to make this conditionally on the python version in use, so stay tuned for an update, without the py3.5 drop.

File: git/cmd.py
Comment (Byron) [inline git/cmd.py:908]: In order to make people aware, could you add a deprecation warning mentioning the date until this will work?

**Comment by Byron** [comment]
Thanks a lot for your help!

I will set a reminder for myself to remove python 3.5 support on September 5th. It would be super helpful to have a PR waiting which does that - the code was present here but seems to have been overwritten.

**Pull request merged**
